proposals/p5168.md

```diff
@@ -449,23 +449,22 @@ with the following updated rules:
     the `impl` definition. At that point, each associated constant that has not
     been assigned a value is given its default value. If it does not have a
     default, an error diagnostic is issued.
--   The associated function signatures are fixed by the end of the impl
-    definition. Associated functions not mentioned in the `impl` definition are
-    given the default signature from the interface.
-    -   This allows `impl` definitions to be short and reduces repetition.
-        Reduced repetition means fewer things to update when something changes.
+-   The associated function declarations are fixed by the end of the `impl`
+    definition.
+    -   Only associated functions with defaults may be omitted from the `impl`
+        definition.
+    -   Not mentioning an associated function with a default in the `impl`
+        definition means the default version from the interface will be used.
+    -   Associated functions declared in the `impl` definition must have a
+        matching definition.
+    -   If the declared signature of an associated function in the `impl`
+        definition does not match the signature given in the `interface`
+        definition, but are compatible, a stub function that does the conversion
+        is generated to bridge between the two versions.
 -   Out-of-line function bodies are allowed in the impl file, even when the
     `impl` definition is in the api file.
-    -   If an out-of-line function signature does not match the signature given
-        by the `impl` definition, but are compatible, a stub function that does
-        the conversion is generated to bridge between the two versions.
     -   This parallels how function definitions are not part of the definition
         of classes.
--   An out-of-line definition of a function may subsume a default function
-    definition from the interface. This may be resolved at link time if the
-    function is defined in a different file.
-    -   An `impl` definition only fixes associated function signatures, not
-        function definitions.
 -   Missing function body definitions are diagnosed at link time.
```

Comment:
I updated this section to reflect the change that we require impl definitions to contain associated function declarations.